packages/manager/apps/pci-private-network/src/pages/new/button-action/ButtonAction.component.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import {
+  OsdsButton,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+import {
+  ODS_BUTTON_VARIANT,
+  ODS_BUTTON_TYPE,
+  ODS_SPINNER_SIZE,
+} from '@ovhcloud/ods-components';
+import { isDiscoveryProject, useProject } from '@ovh-ux/manager-pci-common';
+import { useNavigate } from 'react-router-dom';
+import { useFormContext } from 'react-hook-form';
+import { ErrorResponse } from '@/types/network.type';
+import { ROUTE_PATHS } from '@/routes';
+import { NewPrivateNetworkForm } from '@/types/private-network-form.type';
+import {
+  createPrivateNetwork,
+  refreshPrivateNetworkList,
+} from '@/data/services/services';
+
+const ButtonAction: React.FC = () => {
+  const { t } = useTranslation(['new', 'common']);
+  const { addError, addSuccess } = useNotifications();
+
+  const form = useFormContext<NewPrivateNetworkForm>();
+
+  const { data: project } = useProject();
+  const projectId = useMemo(() => project.project_id, [project]);
```

Comment:
```suggestion
  const projectId = project.project_id;
```
No need for useMemo here since it's only O(1) complexity. Also it seems to be used only one time so maybe it is not necessary